androidApp/build.gradle.kts

```diff
@@ -33,7 +39,20 @@ sentry {
     org = "mbtace"
     projectName = "mobile_app_android"
     authToken = System.getenv("SENTRY_AUTH_TOKEN")
-    autoInstallation { sentryVersion = "8.15.1" }
+    autoInstallation {
+        sentryVersion = provider {
+            val bareKMPConfig = configurations.detachedConfiguration(libs.sentry.kmp.get())
+            val resolvedDependencies = bareKMPConfig.incoming.resolutionResult
+            val resolvedModuleVersions =
+                resolvedDependencies.allComponents.mapNotNull { it.moduleVersion }
+            val transitiveSentryCore =
+                resolvedModuleVersions.find { it.module.toString() == "io.sentry:sentry" }
+            checkNotNull(transitiveSentryCore) {
+                    "Could not find io.sentry:sentry among ${resolvedModuleVersions.joinToString(prefix = "[", postfix = "]")}"
+                }
+                .version
```

Comment:
I hate Gradle. I tried like a dozen times to force both `shared` and `androidApp` to use the latest version of the Sentry Java SDK, but none of the inexplicably numerous ways to do that that the Gradle docs mention were working, so the best we can do is detecting what Sentry Java version Sentry KMP is depending on and using that as the Sentry Java version to install via the Sentry Android Gradle plugin.

Reply:
https://github.com/getsentry/sentry-kotlin-multiplatform/issues/450